### PR TITLE
EP-16 Delete account

### PR DESCRIPTION
**Pull request by ghiobi**
- delete account test
- implemented in user controller delete account
- add new success response

### CODE REVIEW

File: app/Http/Controllers/Controller.php [hunk 1]
```diff
@@ -36,4 +36,16 @@ protected function buildFailedValidationResponse(Request $request, $errors)
...
+    /**
+     * Returns a json success response.
+     *
+     * @return JsonResponse
+     */
+    protected function successResponse()
+    {
+        return new JsonResponse([
+            'status' => 'ok'
+        ]);
```
Comment (JTalarico) [inline app/Http/Controllers/Controller.php]: Nice job. Before we we're returning just a string "ok". Please keep using JSON responses as these are the proper convention.

Comment (ghiobi) [inline app/Http/Controllers/Controller.php]: yes sir

File: app/Http/Controllers/User/UserController.php [hunk 2]
```diff
@@ -85,12 +86,25 @@ public function update(Request $request)
...
-//
-//    }
+    public function destroy(Request $request)
+    {
+        $user = $request->user();
+
+        if ($user->transfersOut()->where('status', 'pending')->count()) {
+            return $this->buildFailedValidationResponse($request, 'You must complete all transfers.');
+        }
+
```
Comment (JTalarico) [inline app/Http/Controllers/User/UserController.php]: Does this include pending transactions that others have sent you AND transactions that you have sent other users.

What I mean is, what if someone sends you money, deletes there account, and then you decline the transaction. What happens to that money?

Comment (ghiobi) [inline app/Http/Controllers/User/UserController.php]: No this does not include pending transfers going inward. There is not record of transfers going in and is only stored on the client side.

> Anyone who has never made a mistake has never tried anything new. -Albert Einstein

**Pull request merged**